### PR TITLE
pkg/collectors: Refactor endpoint and hpa collector

### PR DESCRIPTION
**Pull request by mxinden**
**What this PR does / why we need it**:

This patch refactors the endpoint and horizontal pod autoscaler collector to the format introduced in #577.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -144,7 +144,20 @@ func TestFullScrapeCycle(t *testing.T) {
 
 	body, _ := ioutil.ReadAll(resp.Body)
 
-	expected := `# HELP kube_pod_info Information about pod.
+	// TODO: Move this to a golden record file?
```
Comment (brancz) [inline main_test.go]: as I mentioned in #590 I don't think the maintenance burden of this is actually worth it, I'd vote for removing this check entirely, as the unit and e2e tests already validate the same, without the additional maintenance required

**Comment by brancz** [comment]
this needs a rebase I think

**Comment by brancz** [comment]
this needs a rebase

**Comment by mxinden** [comment]
@brancz rebased. Would you mind taking another look?

**Comment by mxinden** [comment]
@brancz Resolved final merge commit. Would you mind giving this another :+1: ?

**Pull request merged**